executor/join.go

```diff
@@ -889,12 +913,38 @@ func (e *NestedLoopApplyExec) Next(ctx context.Context, req *chunk.Chunk) (err e
 			e.hasMatch = false
 			e.hasNull = false
 
-			for _, col := range e.outerSchema {
-				*col.Data = e.outerRow.GetDatum(col.Index, col.RetType)
-			}
-			err = e.fetchAllInners(ctx)
-			if err != nil {
-				return err
+			if e.canUseCache {
+				var keyByte []byte
+				for _, col := range e.outerSchema {
+					*col.Data = e.outerRow.GetDatum(col.Index, col.RetType)
+					keyByte, err = codec.EncodeKey(e.ctx.GetSessionVars().StmtCtx, keyByte, *col.Data)
```

Comment:
Why not use HashChunkRow here?

Reply:
Just to convert the `correlated columns' data` to `[]byte` as a cache key. The different `correlated columns' data` should encode to different `[]byte`. If we use `HashChunkRow`, there may be a possibility of hash collision.